cmd/bandd/cmd/init.go

```diff
@@ -150,45 +129,125 @@ func InitCmd(customAppState map[string]json.RawMessage, defaultNodeHome string)
+			appGenesis.InitialHeight = initHeight
+
+			consensusParams := tmtypes.DefaultConsensusParams()
+			consensusParams.Block = tmtypes.BlockParams{
+				MaxBytes: 3000000,
+				MaxGas:   50000000,
 			}
-			toPrint := newPrintInfo(config.Moniker, chainID, nodeID, "", appState)
+			consensusParams.Validator = tmtypes.ValidatorParams{
+				PubKeyTypes: []string{
+					tmtypes.ABCIPubKeyTypeSecp256k1,
+				},
+			}
+
+			appGenesis.Consensus = &types.ConsensusGenesis{
+				Params:     consensusParams,
+				Validators: nil,
+			}
+
+			if err = genutil.ExportGenesisFile(appGenesis, genFile); err != nil {
+				return errorsmod.Wrap(err, "Failed to export genesis file")
+			}
+
 			cfg.WriteConfigFile(filepath.Join(config.RootDir, "config", "config.toml"), config)
+			toPrint := newPrintInfo(config.Moniker, chainID, nodeID, "", appState)
 			return displayInfo(toPrint)
 		},
 	}
-	cmd.Flags().String(cli.HomeFlag, defaultNodeHome, "node's home directory")
+
+	cmd.Flags().String(flags.FlagHome, defaultNodeHome, "node's home directory")
 	cmd.Flags().BoolP(FlagOverwrite, "o", false, "overwrite the genesis.json file")
 	cmd.Flags().Bool(FlagRecover, false, "provide seed phrase to recover existing key instead of creating")
 	cmd.Flags().String(flags.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
-	cmd.Flags().Int(FlagTimeoutCommit, 1500, "timeout commit of the node in ms")
+	cmd.Flags().
+		String(FlagDefaultBondDenom, "", "genesis file default denomination, if left blank default value is 'stake'")
+	cmd.Flags().Int64(flags.FlagInitHeight, 1, "specify the initial block height at genesis")
+
 	return cmd
 }
+
+// InitializeNodeValidatorFilesFromMnemonic creates private validator and p2p configuration files using the given mnemonic.
+// If no valid mnemonic is given, a random one will be used instead.
+func InitializeNodeValidatorFilesFromMnemonic(
+	config *cfg.Config,
+	mnemonic string,
+) (nodeID string, valPubKey cryptotypes.PubKey, err error) {
+	if len(mnemonic) > 0 && !bip39.IsMnemonicValid(mnemonic) {
+		return "", nil, fmt.Errorf("invalid mnemonic")
+	}
+	nodeKey, err := p2p.LoadOrGenNodeKey(config.NodeKeyFile())
+	if err != nil {
+		return "", nil, err
+	}
+
+	nodeID = string(nodeKey.ID())
+
+	pvKeyFile := config.PrivValidatorKeyFile()
+	if err := os.MkdirAll(filepath.Dir(pvKeyFile), 0o777); err != nil {
+		return "", nil, fmt.Errorf("could not create directory %q: %w", filepath.Dir(pvKeyFile), err)
+	}
+
+	pvStateFile := config.PrivValidatorStateFile()
+	if err := os.MkdirAll(filepath.Dir(pvStateFile), 0o777); err != nil {
+		return "", nil, fmt.Errorf("could not create directory %q: %w", filepath.Dir(pvStateFile), err)
+	}
+
+	var filePV *privval.FilePV
+	if len(mnemonic) == 0 {
+		filePV = privval.LoadOrGenFilePV(pvKeyFile, pvStateFile)
+	} else {
+		privKey := tmsecp256k1.GenPrivKeySecp256k1([]byte(mnemonic))
+		filePV = privval.NewFilePV(privKey, pvKeyFile, pvStateFile)
+		filePV.Save()
+	}
```

Comment:
if the user doesn't input mnemonic, it will use privval which generates ed25519 key.